generator/src/generator.ts

```diff
@@ -1669,6 +1669,47 @@ export class Generator {
     if (this.combineSubpaths) map.combineSubpaths();
     return map.toJSON();
   }
+
+  /**
+   *
+   * @param outputDirectory The output directory, resolved relative to process.cwd(), where the import
+   * map and its files are stored on the file system. Defaults to 'dist'
+   * @param importMapFileName The name of the import map file stored to disk, within the outputDirectory. Defaults to
+   * main.importmap
+   * @param hostedBaseUrl a fully qualified or relative string path used as base url for all URLs within the import
+   * If not provided, hostedBaseUrl defaults to './'
+   */
+  async downloadAll(
+    outputDirectory: string = 'deps',
+    importMapFileName: string = 'main.importmap',
+    hostedBaseUrl: string = './'
+  ) {
+    if (!isNode) {
+      throw new JspmError('downloadAll is only available in nodejs');
+    }
+
+    const fs = await import('node:fs/promises');
+    const path = await import('node:path');
+
+    for (const url in this.traceMap.installer.resolver.traceEntries) {
```

Comment:
There should be a provider hook to get all the files of a package now I think - https://github.com/jspm/jspm/blob/main/generator/src/providers/index.ts#L45. It's not fully integrated yet, but that would be a nice way to capture this one.

Reply:
For iterating the packages, perhaps we could have a `generator.getPackageList` helper method to extract the package list from the generator of the form `{ registry, name, version, layer }[]`.

Reply:
There were two feedback comments here:

1. Using a provider hook to get the list of files for a package to download
2. Extracting out `generator.getPackageList()` as an intermediate method to find the canonical package list used by the current import map.

I'm fine with skipping (2) above, but I do think we need to solve (1) for this PR, otherwise only downloading partial packages risks paths like asset loading and dynamic import being missed.